ash/src/entry.rs

```diff
@@ -238,11 +238,18 @@ impl Entry {
     /// <https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkEnumerateInstanceExtensionProperties.html>
     pub fn enumerate_instance_extension_properties(
         &self,
+        p_layer_name: Option<&CStr>,
     ) -> VkResult<Vec<vk::ExtensionProperties>> {
         unsafe {
             read_into_uninitialized_vector(|count, data| {
-                self.entry_fn_1_0
-                    .enumerate_instance_extension_properties(ptr::null(), count, data)
+                self.entry_fn_1_0.enumerate_instance_extension_properties(
+                    match p_layer_name {
+                        None => ptr::null(),
+                        Some(str) => str.as_ptr(),
+                    },
```

Comment:
We have `as_raw_ptr` for this:

https://github.com/MaikKlein/ash/blob/bb02b203391d77e30944f4fc90d7a446474de4c7/ash/src/lib.rs#L55-L67
```suggestion
                    layer_name.as_raw_ptr(),
```

Reply:
Welp, thinking about this again this `as_ptr()` is different from a regular borrow of course!

Unfortunately there exists no `.map_or_default()`, but I wonder if the rest of the code uses `.map_or(Default::default(), |s| s.as_ptr())` elsewhere too?